File: JH/DAY12/10866.cpp
#include <bits/stdc++.h>
using namespace std;


deque<int>deck;

int main(){


    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);


    int n;
    cin >> n;

    for (int i = 0; i <n; ++i) {
        string s;
        cin >> s;

        if(s == "push_front"){
            int x;
            cin >> x;
            deck.push_front(x);
        }
        if(s == "push_back"){
            int x;
            cin >> x;
            deck.push_back(x);
        }
        if(s == "pop_front"){
            if(deck.size()==0){
                cout << "-1" << "\n";
            }else{
                cout << deck.front() << "\n";
                deck.pop_front();
            }
        }
        if(s == "pop_back"){
            if(deck.size()==0){
                cout << "-1" << "\n";
            }else{
                cout << deck.back() << "\n";
                deck.pop_back();
            }
        }
        if(s == "size"){
            cout << deck.size()<<"\n";
        }
        if(s == "empty"){
            if(deck.empty()){
                cout << "1" <<"\n";
            }else{
                cout << "0" << "\n";
            }
        }
        if(s == "front"){
            if(deck.size()==0){
                cout << "-1" << "\n";
            }else{
                cout << deck.front() << "\n";
            }
        }
        if(s == "back"){
            if(deck.size()==0){
                cout << "-1" << "\n";
            }else{
                cout << deck.back() << "\n";
            }
        }

    }





}

File: KSY/DAY13/10866.py
import sys
from collections import deque

input = sys.stdin.readline
n = int(input())
deq = deque()

for _ in range(n):
    query = list(input().split())

    if query[0] == 'push_back':
        deq.append(int(query[1]))
    elif query[0] == 'push_front':
        deq.appendleft(int(query[1]))
    elif query[0] == 'front':
        if deq:
            print(deq[0])
        else:
            print(-1)
    elif query[0] == 'back':
        if deq:
            print(deq[-1])
        else:
            print(-1)
    elif query[0] == 'size':
        print(len(deq))
    elif query[0] == 'empty':
        if not deq:
            print(1)
        else:
            print(0)
    elif query[0] == 'pop_front':
        if deq:
            print(deq.popleft())
        else:
            print(-1)
    elif query[0] == 'pop_back':
        if deq:
            print(deq.pop())
        else:
            print(-1)